### PR TITLE
v2: add doctests and tests for example scripts

### LINKED ISSUE #249: add doctest for python code in documentation
We are using markdown for documentation in v2. The documentation contains many code examples that should be tested for correctness. Add doctest for python code in the markdown documentation.

An option to consider: https://pypi.org/project/phmdoctest/

### PR DESCRIPTION
**Pull request by jreiberkyle**
This work provides nox sessions (not run by default) for running doctests in markdown files and for running all scripts in the examples/ folder to ensure they run successfully. As a part of this work, the long example in README.md is moved to `exampes/orders_create_and_download_multiple_orders.py` and a README is provided for the examples folder that give some insight into how examples can be discovered, run, and developed. Additionally, `test_examples.py` was created to automatically test the scripts with pytest.

Closes #249

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Navigation
+Each example script is named according to the convention
+`<Planet API>_<use_case>.py`.
+
+## Usage
+
+To use these examples, just run the scripts with `python <script_name>`. These
+examples hit the Planet servers, so be aware that usage and fees may apply.
+
+## Contributing
```
Comment (ericrdunham) [inline examples/README.md:14]: Oooh, this might be nice to add to a CONTRIBUTING.md!

Comment (jreiberkyle) [reply examples/README.md:14]: Great suggestion. Added a reference to this doc in CONTRIBUTING.md That prompted a whole update of CONTRIBUTING.md, which brings it up to date with nox usage and testing docs/examples. Take a look if you get a chance.

File: examples/test_examples.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test example scripts in this directory
+
+This is not an example script. This is a pytest module that provides automated
+testing of the example scripts in this directory to ensure they are always
+up-to-date and run successfully.
+
+YAY automated testing and up-to-date examples!
```
Comment (sarasafavi) [inline examples/test_examples.py:20]: Sweet! LGTM, this aligns with our previous discussions on the topic outside of GH

**Comment by sarasafavi** [comment]
This all LGTM; these updates align with our previous discussions on the topic outside of GH.

**Review by sarasafavi — APPROVED** [review]

**Review by ericrdunham — COMMENTED** [review]
Thanks for cleaning up the CONTRIBUTING.md! :)

**Pull request merged**